### PR TITLE
Sanitizer/Valgrind CI

### LINKED ISSUE #412: CI memory checks
GHA additions:
- [ ] Address/leak sanitizer
- [ ] valgrind

### PR DESCRIPTION
**Pull request by junghans**
First part of #412

### CODE REVIEW

**Comment by junghans** [comment]
@sslattery vlagrind works and found a couple of things!

Memory and address sanitizer still needs to enabled a a build.

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,6 +152,26 @@ if(Cabana_ENABLE_COVERAGE_BUILD)
...
+endif()
+if(WITH_ASAN)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+endif()
+
+if(WITH_MSAN)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=memory -fno-omit-frame-pointer")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=memory")
+endif()
```
Comment (dalg24) [inline CMakeLists.txt:173]: Why an option when all you need is CXX flags?

Comment (junghans) [reply CMakeLists.txt:173]: Only clang support `-fsanitize=memory` and some obscure compilers might not have `-fsanitize=address`.

**Comment by junghans** [comment]
@sslattery @streeve I don't really like the fact that we are overwriting the existing tests, I think it would be better to create a second test for valgrind instead.

Any opinions?

**Comment by junghans** [comment]
@streeve should be use enable `-fsanitize=address` all the time and add one build for `-fsanitize=memory`, or what do you think?

**Comment by streeve** [comment]
So `-fsanitize=address` all the time and one additional `-fsanitize=memory` and one valgrind? 

That sounds good to me

**Comment by streeve** [comment]
Couple of failures for the memory sanitizer to fix, but the address sanitizer looks good

**Comment by streeve** [comment]
Okay @junghans this is passing with only address santizer @sslattery anything you want to change for these newoptions?

**Review by sslattery — APPROVED** [review]

**Pull request merged**